libs/domains/variables/feature/src/lib/variable-list/variable-list.tsx

```diff
@@ -0,0 +1,408 @@
+import {
+  type SortingState,
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  getFacetedRowModel,
+  getFacetedUniqueValues,
+  getFilteredRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from '@tanstack/react-table'
+import {
+  APIVariableScopeEnum,
+  type APIVariableTypeEnum,
+  LinkedServiceTypeEnum,
+  type VariableResponse,
+} from 'qovery-typescript-axios'
+import { Fragment, useMemo, useState } from 'react'
+import { NavLink } from 'react-router-dom'
+import { match } from 'ts-pattern'
+import { ExternalServiceEnum, IconEnum } from '@qovery/shared/enums'
+import { APPLICATION_GENERAL_URL, APPLICATION_URL, DATABASE_GENERAL_URL, DATABASE_URL } from '@qovery/shared/routes'
+import {
+  Button,
+  DropdownMenu,
+  EmptyState,
+  Icon,
+  PasswordShowHide,
+  TableFilter,
+  TablePrimitives,
+  Tooltip,
+} from '@qovery/shared/ui'
+import { dateUTCString, timeAgo } from '@qovery/shared/util-dates'
+import {
+  environmentVariableFile,
+  getEnvironmentVariableFileMountPath,
+  pluralize,
+  twMerge,
+  upperCaseFirstLetter,
+} from '@qovery/shared/util-js'
+import { useVariables } from '../hooks/use-variables/use-variables'
+import { VariableListSkeleton } from './variable-list-skeleton'
+
+const { Table } = TablePrimitives
+
+export interface VariableListProps {
+  className?: string
+  showAll?: boolean
+  parentId: string
+  currentScope: keyof typeof APIVariableScopeEnum
+  onCreateVariable: (
+    variable: VariableResponse,
+    variableType: typeof APIVariableTypeEnum.OVERRIDE | typeof APIVariableTypeEnum.ALIAS
+  ) => void
+  onEditVariable: (variable: VariableResponse) => void
+  onDeleteVariable: (variable: VariableResponse) => void
+}
+
+export function VariableList({
+  className,
+  showAll = false,
+  parentId,
+  currentScope,
+  onCreateVariable,
+  onEditVariable,
+  onDeleteVariable,
+}: VariableListProps) {
+  const { data: variables = [], isLoading: isVariablesLoading } = useVariables({
+    parentId,
+    scope: currentScope,
+  })
+  const [sorting, setSorting] = useState<SortingState>([])
+
+  const organizationId = ''
+  const projectId = ''
+  const environmentId = ''
+
+  const columnHelper = createColumnHelper<(typeof variables)[number]>()
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor('key', {
+        id: 'key',
+        header: `${variables.length} ${pluralize(variables.length, 'variable')}`,
+        enableColumnFilter: false,
+        size: 40,
+        cell: (info) => {
+          const variable = info.row.original
+          return (
+            <>
+              {variable.owned_by === ExternalServiceEnum.DOPPLER && (
+                <span
+                  data-testid="doppler-tag"
```

Comment:
```suggestion
                  data-testid="doppler-tag"
```
useless ?

Reply:
It exists in the old version, I just kept it to be iso